a10_neutron_lbaas/handler_base.py

```diff
@@ -13,15 +13,25 @@
 #    under the License.
 
 import json
+import logging
+
+LOG = logging.getLogger(__name__)
 
 
 class HandlerBase(object):
 
     def __init__(self, a10_driver):
         self.a10_driver = a10_driver
-        self.hooks = a10_driver.hooks
         self.openstack_driver = self.a10_driver.openstack_driver
 
+        # XXX(dougwig): is this too racy?
+        if self.a10_driver.provider is None:
+            for prov, driver in self.openstack_driver.plugin.drivers.items():
+                if driver is self.openstack_driver:
+                    self.a10_driver._late_init(prov)
```

Comment:
I think provider, config, and hooks all move to handler_base, actually, and out of global init entirely.